src/Microsoft.DotNet.XHarness.Tests.Runners/NUnit/NUnitTestRunner.cs

```diff
@@ -0,0 +1,172 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Common;
+using Microsoft.DotNet.XHarness.Tests.Runners.Core;
+using NUnit.Engine;
+
+#nullable enable
+namespace Microsoft.DotNet.XHarness.Tests.Runners.NUnit
+{
+	internal class NUnitTestRunner : TestRunner, INUnitTestRunner
+	{
+        private readonly FilterBuilder _testFilterBuilder;
+        private readonly NUnitTestListener _testListener;
+		private ResultSummary? _results;
+		private bool _runAssemblyByDefault;
+
+        private  Dictionary<string, bool>? AssemblyFilters { get; set; }
+
+		protected override string ResultsFileName { get; set; } = "TestResults.NUnit.xml";
+
+		public bool GCAfterEachFixture { get; set; }
+
+		public NUnitTestRunner (LogWriter logger) : base (logger)
+		{
+            _testListener = new NUnitTestListener(runner: this, logger: logger);
+            _testFilterBuilder = new FilterBuilder(new TestFilterBuilder());
+		}
+
+		public override async Task Run (IEnumerable<TestAssemblyInfo> testAssemblies)
+		{
+			if (testAssemblies == null)
+				throw new ArgumentNullException (nameof (testAssemblies));
+
+			if (AssemblyFilters == null || AssemblyFilters.Count == 0)
+				_runAssemblyByDefault = true;
+			else
+				_runAssemblyByDefault = AssemblyFilters.Values.Any (v => !v);
```

Comment:
Somehow I feel like you could use `!AssemblyFilters.Values.All()` but not sure I got it right